File: todo-reservation-springboot/src/main/java/de/vkb/dojo/es/humanResources/model/event/PersonDeleted.java
package de.vkb.dojo.es.humanResources.model.event;

import com.fasterxml.jackson.annotation.*;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("person-deleted")
public class PersonDeleted extends PersonEvent {
    private final String operationId;
    private final String aggregateId;

    @JsonCreator
    public PersonDeleted(
            @JsonProperty("operationId") String operationId,
            @JsonProperty("aggregateId") String aggregateId
    ) {
        this.operationId = operationId;
        this.aggregateId = aggregateId;
    }

    @Override
    @JsonGetter
    public String getOperationId() {
        return operationId;
    }


    @Override
    @JsonGetter
    public String getAggregateId() {
        return aggregateId;
    }
}

File: todo-reservation-micronaut/src/main/kotlin/de/vkb/dojo/es/humanResources/model/event/PersonDeleted.kt
package de.vkb.dojo.es.humanResources.model.event

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonTypeName

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("person-deleted")
data class PersonDeleted(
    override val operationId: String,
    override val aggregateId: String
): PersonEvent
